src/openforms/submissions/tests/test_endpoints.py

```diff
@@ -153,3 +154,185 @@ def test_token_valid_on_same_day(self):
         response = self.client.get(endpoint)
 
         self.assertEqual(response.status_code, 302)
+
+    def test_redirects_to_auth_if_form_requires_login(self):
+        form = FormFactory.create()
+        form_step = FormStepFactory.create(
+            form=form, form_definition__login_required=True
+        )
+        submission = SubmissionFactory.create(form=form, auth_plugin="digid")
+        SubmissionStepFactory.create(
+            submission=submission, form_step=form_step, data={"foo": "bar"}
+        )
```

Comment:
think you can shortcut this a lot with either `SubmissionFactory.create(form__generate_minimal_setup=True, form__formstep__form_definition__login_required=True)` or `SubmissionFactory.from_components(...)`